homeassistant/components/tuya/humidifier.py

```diff
@@ -169,10 +202,24 @@ def current_humidity(self) -> int | None:
 
     def turn_on(self, **kwargs: Any) -> None:
         """Turn the device on."""
+        if not self.has_required_dpcodes:
+            raise ServiceValidationError(
```

Comment:
IMO all these raise changes should be in a separate PR, and aligned for all platforms. 
Most still raise RuntimeError and I think it should be changed for all